src/libcore/ffi.rs

```diff
@@ -268,35 +268,86 @@ mod sealed_trait {
                reason = "the `c_variadic` feature has not been properly tested on \
                          all supported platforms",
                issue = "44930")]
-    pub trait VaArgSafe {}
+    pub trait VaArgSafe {
+        #[doc(hidden)]
+        #[cfg(not(bootstrap))]
+        unsafe fn __va_arg(ap: &mut super::VaListImpl<'_>) -> Self;
```

Comment:
I'm not sure about the double-underscore prefix, wouldn't `va_arg_internal` be a better name?

Reply:
It's sealed so `va_arg` or `va_arg_from_list` would be fine.